lib/screens/v2/widget_instance/reconstructed_alert.ex

```diff
@@ -72,183 +139,351 @@ defmodule Screens.V2.WidgetInstance.ReconstructedAlert do
+
+    informed_entities
+    |> Enum.filter(&(&1.route_type in [0, 1] and &1.route in routes_at_stop))
+    |> Enum.group_by(fn %{route: route} -> route end)
+    |> Enum.map(fn
+      {route_id, _} ->
+        headsign = get_destination(t, location, route_id)
+
+        if is_nil(headsign) do
+          route_id |> String.first() |> String.downcase() |> Kernel.<>("l")
+        else
+          format_for_svg_name(headsign)
+        end
+    end)
+    |> Enum.uniq()
+  end
+
+  defp format_for_svg_name("Ashmont/Braintree"),
+    do: [Map.get(@headsign_svg_map, "Ashmont"), Map.get(@headsign_svg_map, "Braintree")]
+
+  defp format_for_svg_name(headsign), do: Map.get(@headsign_svg_map, headsign)
+
+  defp format_cause(:unknown), do: nil
+  defp format_cause(cause), do: cause |> to_string() |> String.replace("_", " ")
+
+  defp format_routes(routes) do
+    Enum.map(routes, fn
+      "Green-" <> branch ->
+        "gl-#{String.downcase(branch)}"
+
+      route_id ->
+        String.downcase(route_id)
+    end)
+  end
+
+  defp get_region_from_location(:inside), do: :here
+
+  defp get_region_from_location(location)
+       when location in [:boundary_upstream, :boundary_downstream],
+       do: :boundary
+
+  defp get_region_from_location(_location), do: :outside
+
+  defp get_cause(:unknown), do: nil
+  defp get_cause(cause), do: cause
+
+  def takeover_alert?(%__MODULE__{is_full_screen: false}), do: false
+
+  def takeover_alert?(%__MODULE__{is_terminal_station: is_terminal_station, alert: alert} = t) do
     Alert.effect(alert) in [:station_closure, :suspension, :shuttle] and
       LocalizedAlert.location(t, is_terminal_station) == :inside and
-      LocalizedAlert.informs_all_active_routes_at_home_stop?(t)
+      LocalizedAlert.informs_all_active_routes_at_home_stop?(t) and
+      (is_nil(Alert.direction_id(t.alert)) or is_terminal_station)
   end
 
+  @spec serialize_takeover_alert(t()) :: takeover_serialized_response()
+  defp serialize_takeover_alert(t)
+
   defp serialize_takeover_alert(
          %__MODULE__{
-           alert: %Alert{effect: :suspension, cause: cause} = alert
+           alert: %Alert{effect: :suspension, cause: cause, updated_at: updated_at} = alert,
+           now: now
          } = t
        ) do
     informed_entities = Alert.informed_entities(alert)
-    affected_routes = LocalizedAlert.informed_subway_routes(t)
-    cause_text = cause |> Alert.get_cause_string() |> String.capitalize()
-
-    location_text = get_endpoints(informed_entities, hd(affected_routes))
-
-    issue = %FreeTextLine{
-      icon: nil,
-      text:
-        ["No"] ++
-          (affected_routes
-           |> Enum.map(fn route ->
-             %{
-               route:
-                 route
-                 |> String.replace("-", "_")
-                 |> String.downcase()
-             }
-           end)
-           |> Enum.to_list()) ++
-          ["trains"]
-    }
+    [route_id] = LocalizedAlert.informed_subway_routes(t)
```

Comment:
My assumption that it will only contain one route stems from that being unconventional for a suspension. 

At the moment, we only have one screen on the GL, and we cannot show a takeover alert for a GL suspension there. This will be true until we have a way to infer that a transfer station has no service based on a combination of alerts (so GL suspension and BL shuttle or something similar).